### PR TITLE
S3 parallel copy state machine app

### PR DESCRIPTION
**Pull request by ttung**
Connects to #592.
Depends on #704, #707.

### CODE REVIEW

File: iam/policy-templates/dss-s3-copy-sfn-lambda.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      "Action": [
+        "logs:CreateLogGroup",
+        "logs:CreateLogStream",
+        "logs:PutLogEvents"
+      ],
+      "Resource": "arn:aws:logs:*:*:*"
+    },
+    {
+      "Effect": "Allow",
+      "Action": "s3:*",
```
Comment (kislyuk) [inline iam/policy-templates/dss-s3-copy-sfn-lambda.json]: This lambda shouldn't have delete access to the bucket. That's dangerous. Let's set the permissions on an as-needed basis.

File: daemons/dss-s3-copy-sfn/app.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+sys.path.insert(0, pkg_root)  # noqa
+
+import dss.stepfunctions.s3copyclient as s3copyclient
+import dss.stepfunctions.generator as generator
+
+app = domovoi.Domovoi()
+
+
+transformer = generator.Transformer(app)
+transformer.transform(s3copyclient.sfn)
```
Comment (kislyuk) [inline daemons/dss-s3-copy-sfn/app.py]: Any chance we could use more descriptive names than generator and transformer?

Comment (ttung) [reply daemons/dss-s3-copy-sfn/app.py]: Done.

File: iam/policy-templates/dss-s3-copy-sfn-lambda.json (hunk 1 above)
Comment (ttung) [reply iam/policy-templates/dss-s3-copy-sfn-lambda.json]: Done.

**Pull request merged**